### ISSUE TITLE
[TASK] Add possibility for building statically

### ISSUE THREAD

**Issue by arnout** [prplfoundation/prplMesh#469]
## Description

prplMesh now consists of a many relatively small shared libraries. These have a significant memory footprint. We would like to have the possibility to build static libraries instead, which may reduce the size (especially if it is combined with link-time optimisation).

Instead of forcing shared libraries, allow to build static libraries with the standard CMake option `BUILD_SHARED_LIBS`.

bwl and bpl are currently even dlopen'ed, but there's no need for that. The libraries are not replaced dynamically, they are fixed for a specific platform.

## Exit criteria

- It is possible to build with `-DBUILD_SHARED_LIBS=OFF`
- Analyze executable size difference
- Identify other routes to decrease size

**Comment by vitalybu** [comment]
Memory or disk footprint?
Maybe I'm missing something, but shared libraries usually reduce memory footprint, as they are loaded only once into the memory.
If the BCL for example will link statically to all the executables, they'll take considerably more space both on disk and in-memory.

**Comment by arnout** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Support static linking](https://trello.com/c/gjsv6UWn/35-support-static-linking)

**Comment by arnout** [comment]
An attempt was made in #560 but doesn't completely work. For the time being, we're putting this on the back burner.

**Comment by ghost** [comment]
This is needed for Econet

**Comment by ghost** [comment]
@arnout Can this issue be closed? Or is further work needed?

**Comment by juanschroeder** [comment]
What I did for Econet (econet/econet_static) was to do a full static build, including libc and dependencies. I added a new CMake option BUILD_FULL_STATIC.
Is that what was needed here?

**Comment by arnout** [comment]
I think the BUILD_FULL_STATIC is indeed what what we need to aim for. Only I'm not so sure if we really should include that workaround for the buggy Ubuntu toolchain...

**Comment by arnout** [comment]
Closing - we're currently building with CMAKE_SHARED_LIBS=OFF in CI. Full static building is more of an integration thing - we'll take bugfixes for it if someone encounters a problem, but we don't need a separate task for it.
